backend/controller/account.controller.js

```diff
@@ -82,22 +82,22 @@ router.put("/update/:id", auth, async (req, res) => {
 router.delete("/delete/:id", auth, async (req, res) => {
   try {
     const id = req.params.id;
-
-    // find the account by id
-    const account = await Account.findByIdAndDelete(id);
-    const transactions = await Transaction.find({ accountId: id });
-    const user = await User.findById(req.userId);
-    user.set({ accounts: user.accounts.filter((acc) => acc != id) });
-    await user.save();
-
+    const account = await Account.findById(id).exec();
```

Comment:
As i am already giving you accountid you dont have to get account you can directly use its id do what ever you want this will reduce api response time